### PR TITLE
check setsockopt return

### PR DESCRIPTION
**Pull request by gfx**
See the discussions in https://github.com/h2o/h2o/pull/3003

### CODE REVIEW

**Comment by kazuho** [comment]
Thank you for the PR. I believe this is a duplicate of #3003. Please see there for my review comments.

**Comment by gfx** [comment]
I see! Made them MT-safe: 5257f158c

**Comment by kazuho** [comment]
Thank you for the changes.

What do you think about https://github.com/h2o/h2o/pull/3003#pullrequestreview-950943343?

File: src/main.c [hunk 1]
```diff
@@ -1755,11 +1763,7 @@ static void set_quic_sockopts(int fd, int family, unsigned sndbuf, unsigned rcvb
         break;
     }
 
-    /* set sndbuf & rcvbuf */
-    if (sndbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) != 0)
-        h2o_fatal("failed to set SO_SNDBUF:%s", strerror(errno));
-    if (rcvbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_RCVBUF, &rcvbuf, sizeof(rcvbuf)) != 0)
-        h2o_fatal("failed to set SO_RCVBUF:%s", strerror(errno));
+    set_socket_buffers(fd, sndbuf, rcvbuf);
```
Comment (kazuho) [inline src/main.c:1766]: Yeah, these lines of code are executed during startup, therefore we are refusing to boot if they fail.

That's the intent of the code, we cannot change the behavior.

Comment (gfx) [reply src/main.c:1766]: Right! So extracting `set_socket_buffers()` does not make sense. 

The original patch intends to check the call of `setsockopt()`, not sharing the code with elsewhere, so I've made another one: https://github.com/h2o/h2o/pull/3116

**Comment by gfx** [comment]
Closing in favor of https://github.com/h2o/h2o/pull/3116

**Pull request closed without merging**